### PR TITLE
Add decimal support

### LINKED ISSUE #172: Add decimal support
Currently decimals as registered as floats (C# doubles) which leads to an exception when they are used. There is an existing DecimalGraphType from GraphQL.net which should be used for decimals

### PR DESCRIPTION
**Pull request by K-Pavlov**
fixes #172

### CODE REVIEW

File: test/Tests/Execution/SchemaExecutionTests.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    public class SchemaExecutionTests
+    {
+        [Test]
+        public void Can_Have_Decimals_In_Schema()
+        {
+            var schema = SchemaBuilderHelpers.Schema<SchemaTypeWithDecimal>();
+            schema.ShouldHaveQueries(1);
+            schema.ShouldHaveMutations(0);
+            schema.Query.ShouldHaveFieldWithName("test");
+            var res = schema.Execute((e) => e.Query = "query { test }");
```
Comment (tlil) [inline test/Tests/Execution/SchemaExecutionTests.cs]: Test the result as well?

File: test/Tests/SchemaBuilderHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using GraphQL.Conventions.Adapters;
+using GraphQL.Conventions.Builders;
+using GraphQL.Types;
+
+namespace GraphQL.Conventions.Tests
+{
+    public static class SchemaBuilderHelpers
```
Comment (tlil) [inline test/Tests/SchemaBuilderHelpers.cs]: See comment further up - this refactor seems unnecessary.

File: test/Tests/Execution/SchemaExecutionTests.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+using GraphQL.Conventions.Tests.Templates.Extensions;
+
+namespace Tests.Execution
+{
+    public class SchemaExecutionTests
+    {
+        [Test]
+        public void Can_Have_Decimals_In_Schema()
+        {
+            var schema = SchemaBuilderHelpers.Schema<SchemaTypeWithDecimal>();
```
Comment (tlil) [inline test/Tests/Execution/SchemaExecutionTests.cs]: Without the refactoring done further down, you can simply have `SchemaExecutionTests` derive from `ConstructionTestBase` and use `Schema<SchemaTypeWithDecimal>()` here.

**Review by tlil — CHANGES_REQUESTED** [review]
Just a couple of things, otherwise looks good. Thanks for sorting!

**Comment by K-Pavlov** [comment]
Check, will resolve them tommorow

**Review by tlil — APPROVED** [review]

**Comment by shoe-diamente** [comment]
Is there some pre-release / nightly package available that includes this fix?

**Pull request merged**